src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -280,28 +238,31 @@ private void SetupCommonOptionsForCommands(List<Command> commands)
             return (int)exitCode;
         }
 
-        private ExitCode AnalyzeTemplate(FileInfo templateFilePath, FileInfo parametersFilePath)
+        private ExitCode AnalyzeTemplate(TemplateAndParams templateAndParameters)
         {
             try
             {
-                string templateFileContents = File.ReadAllText(templateFilePath.FullName);
-                string parameterFileContents = parametersFilePath == null ? null : File.ReadAllText(parametersFilePath.FullName);
 
-                IEnumerable<IEvaluation> evaluations = this.templateAnalyzer.AnalyzeTemplate(templateFileContents, templateFilePath.FullName, parameterFileContents);
+                (string template, string parameters) = TemplateDiscovery.GetTemplateAndParameterContents(templateAndParameters);
+
+                IEnumerable<IEvaluation> evaluations = this.templateAnalyzer.AnalyzeTemplate(template, templateAndParameters.Template.FullName, parameters);
 
-                this.reportWriter.WriteResults(evaluations, (FileInfoBase)templateFilePath, (FileInfoBase)parametersFilePath);
+                this.reportWriter.WriteResults(evaluations, (FileInfoBase)templateAndParameters.Template, (FileInfoBase)templateAndParameters.Parameters);
 
                 return evaluations.Any(e => !e.Passed) ? ExitCode.Violation : ExitCode.Success;
             }
             catch (Exception exception)
             {
-                if (parametersFilePath != null)
+                // Keeping separate LoggError calls so formatting can use the recommended templating.
+                // https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2254
```

Comment:
If you have time would you mind maybe moving/copying that link to the logger class as a general recommendation?

Reply:
Do you have a location in mind, or recommended way to include the recommendation?

Reply:
Also we have multiple loggers.  Not sure where best to add it.

Reply:
I was thinking under one of these files: https://github.com/Azure/template-analyzer/tree/598b9633515d64f34fb01cd16740900ca5c5b6bf/src/Analyzer.Cli/Loggers but I think you're right, now I don't see where it would fit properly